### ISSUE TITLE
[feature-request] Allow scheduled enable/disable for notifications

### ISSUE THREAD

**Issue by kali-hernandez-sociomantic** [NagiosEnterprises/nagioscore#52]
As described in the support forums (http://support.nagios.com/forum/viewtopic.php?f=7&t=33376&p=142286) it would be nice to have the same functionality already existing in scheduled downtime for checks replicated for notifications disable.

As explained in the link above, some times we disable the notifications temporarily to run some tests and avoid spamming our users, but we later forget to re-enable them. 

It seems to me like it would be fairly easy to apply the same logic as for scheduled downtimes also for the notifications disable feature, thus allowing us to set a specified amount of time for disabling alerts within a given timeframe, letting itself reactivate later.

**Comment by tmcnag** [comment]
+1, would make testing and staging easier in cases where the checks still need to run.

**Comment by box293** [comment]
+1 as it has caught me out in the past as well.

**Comment by tmcnag** [comment]
+1 from the forums:  https://support.nagios.com/forum/viewtopic.php?t=42392

**Comment by jfrickson** [comment]
We will need to add one or more new objects to get this to work properly. So it will have to wait for core 5.

**Comment by ericloyd** [comment]
An NMI type of notification that can't be disabled would also be a good thing(tm).

**Comment by kelemvor33** [comment]
Yes please.  When we do Windows Patching or other things that are going to take down our entire environment for a bit, I'd love to have a way to put our entire nagios system into "Maintenance Mode" for a predetermined number of minutes/hours so that we don't have to remember to go turn alerting back on later as we've had people forget in the past.

**Comment by hedenface** [comment]
Will review internally - but it's obviously a pretty Good Idea. Tentative scheduling for 5.0.0

**Comment by ericloyd** [comment]
That's how I see it.  I mean, you can do this with cron and write to the command file but something baked in would be better.  Ideally with a notification type that can't be disabled for that "last ditch effort" to notify about catastrophic errors.
